### PR TITLE
 Buyer should be able to provide a review#89

### LINKED ISSUE #89: Feat:  Buyer should be able to provide a review
---
### Description

As a buyer, I can submit reviews after receiving a product.

### Tasks

- [x] Create   database model for review  
- [x] Implement an endpoint of posting a review and  updating the product's average rating.
- [x] Implement validation checks  to ensure that reviews meet certain  criteria. and Restrict 
         who can submit review likebuyer must be authorized  
- [x] Write appropriate unit tests
- [x] Write swagger documentation for the resulting endpoints

### Related Issues

#83 

### Definition of Done
- Endpoint of posting a review and  updating the product's average rating.
- Database model for review 
- Unit tests are written and passing
- Swagger documentation

### PR DESCRIPTION
**Pull request by Dawaic6**
### Title: Implement review feature for products

**Description:**

This pull request adds functionality to manage reviews for products like post review and updating averageRate within the application.

#### Implemented Features:
-  Review Endpoint
- unit tests for the review-related endpoints
- swagger document  for Review Endpoint

#### Definition of Done:

-Review Endpoint:

   1.  Create/Post review endpoint
   2. get reviews

-unit tests for the review-related endpoints to ensure the functionality works as expected.
-swagger document  for Review Endpoint

Related Issues:
#83

### To use the Code ###
- **Clone this repository**:
  git clone https://github.com/atlp-rwanda/dynamites-ecomm-be.git
- **Run**:
`npm run dev`
- **Test-covarage**:
`npm run test:ci`
- **Test**:
`npm run test`
### Screenshot
-**for test**
image

-**for swagger**
Screenshot 2024-05-28 182943

### CODE REVIEW

**Review by ambroisegithub — APPROVED** [review]
It looks Good for merge!!!

**Review by jkarenzi — APPROVED** [review]

**Review by bertrandshema — APPROVED** [review]
good

File: src/config/db.ts [hunk 1]
```diff
@@ -28,14 +28,14 @@ const staging = {
   password: process.env.DB_PASSWORD,
   host: process.env.DB_HOST,
   port: process.env.DB_PORT,
-  name: process.env.DB_NAME,
+  name: process.env.DB_NAME_DEV,
```
Comment (dushimeemma) [inline src/config/db.ts:31]: why have you changed this environment variable?

**Review by wayneleon1 — APPROVED** [review]
Look Good

File: src/config/db.ts [hunk 2]
```diff
@@ -28,14 +28,14 @@ const staging = {
...
 };
 const production = {
   username: process.env.DB_USER,
   password: process.env.DB_PASSWORD,
   host: process.env.DB_HOST,
   port: process.env.DB_PORT,
-  name: process.env.DB_NAME,
+  name: process.env.DB_NAME_DEV,
```
Comment (dushimeemma) [inline src/config/db.ts:38]: why have you changed this environment variable?

File: src/config/db.ts (hunk 1 above)
Comment (Dawaic6) [reply src/config/db.ts:31]: oky , leo has make mistake in last PR  so he asked me to change  by avoid to creating PR of that

File: src/config/db.ts (hunk 2 above)
Comment (Dawaic6) [reply src/config/db.ts:38]: also here

**Review by 13XAVI — APPROVED** [review]
cool

**Review by dushimeemma — APPROVED** [review]

**Pull request merged**